web/app.js

```diff
@@ -864,12 +864,29 @@ const PDFViewerApplication = {
       .catch(downloadByUrl); // Error occurred, try downloading with the URL.
   },
 
-  _recordFallbackErrorTelemetry(featureId) {
-    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("MOZCENTRAL")) {
+  /**
+   * For PDF documents that contain e.g. forms and javaScript, we should only
+   * trigger the fallback bar once the user has interacted with the page.
+   * @private
+   */
+  _delayedFallback(featureId) {
+    if (
+      typeof PDFJSDev === "undefined" ||
+      PDFJSDev.test("MOZCENTRAL || GENERIC")
```

Comment:
Telemetry is only implemented (and useful?) for `MOZCENTRAL` as far as I can tell, so why is `GENERIC` added as well?

Reply:
To actually get this code to run in more builds, and thus make it slightly better tested, since having functionality which is important in mozilla-central and which is currently only run in the development viewer seems suboptimal to me.

Originally the `FirefoxCom`-calls were placed inline throughout the viewer code, using pre-processor statements, but at this point there's really nothing stopping us from simply running this kind of code everywhere thanks to dummy methods on `DefaultExternalServices`. Should we perhaps just attempt to report telemetry *unconditionally* in all builds instead?

Reply:
Yes, given that the dummies are in place, I have no problem running this unconditionally.